### DISCUSSION TITLE
Why the name was changed?

### DISCUSSION THREAD

**Discussion by dolfinus** [Neoteroi/BlackSheep#299] [General]
`blacksheep` was quite a unique and funny name for the project.
`neoteroi` is hard to remember and to pronounce.

**Comment by Yourun-proger** [comment]
I completely agree. "Rebranding" turned out to be unsuccessful. Lost uniqueness. It's like renaming flask, marshmallow or pydantic

**Reply by RobertoPrevato** [reply]
Hi! Thank You for your recommendations and interesting input. I've been conflicted about this for **months** and I can still change mind. No damage done yet, because I didn´t publish anything stable to PyPi (only _alpha_ packages). I welcome this discussion.

The main reasons I planned to rename the project are:
1. when there is the opportunity, I like to encapsulate sets of features into dedicated libraries, keeping them agnostic of each other. For example `rodi` for dependency injection, `guardpost` for authentication and authorization part. Over time, I started thinking that this approach doesn´t scale well: I don´t like very much to come up with new names every time I wish to share something. For example I never truly liked the name "guardpost". I chose the name "rodi" for a game of word: the famous Greek island of Rodos in Italian is called "Rodi" - which ends with "di" that is used for dependency injection. Somebody asked me if it was a reference to my name, Roberto, and I sometimes wondered if people think I am eccentric, inserting my own name in a package name.
I thought that using a common namespace and adopting PEP 420 would offer better naming, and clarify that these libraries are coming from the same author / set of libraries.
2. I thought that having more consistent names across the libraries would be, albeit more boring, more professional looking.
3. I was discouraged by some opinions I saw, that having animals in project names is silly and looks unprofessional. I know this is irrational because Mozilla **Firefox** or **Thunderbird** never looked silly or unprofessional to me because of having animals in their names / logos.

But if enough people encourage me to keep the current names for the projects, I'm glad to revert my changes and retrace my steps.

**Reply by Yourun-proger** [reply]
Thank you for quick response! I hope that the truth will be able to keep such beautiful original names. But on the other hand, I can understand why you want names like "X-web", "X-di". But specifically the name "Neoteroi" does not look very good, because, as the OP said, it is difficult to pronounce. So if you do not change your mind about the general concept of the name of your projects, then IMHO I would come up with a new, concise name. In any case, I and many of my friends think that your projects are very useful and look worthy, so good luck with your current projects and creating new ones!

**Reply by RobertoPrevato** [reply]
Thanks for your kind words - they mean a lot to me. When it comes to pronunciation, the word `neoteroi` is Greek and would be pronounced like _nè·oteree_. It means _youngers_ and it was used to describe, scorning, the ["new poets"](https://www.unrv.com/culture/catullus.php) in the 1st century BC. The [English word "neoteric"](https://www.merriam-webster.com/dictionary/neoteric) originates from the same - Merriam gives the English pronounciation as [nē-ə-ˈter-ik](https://www.merriam-webster.com/dictionary/neoteric?pronunciation&lang=en_us&dir=n&file=neoter01). An English influenced pronunciation might be "nē-ə-ˈter-hoi".
As a side note, I don't think it's harder than "Kubernetes" (this is also a Greek word that most people pronounce, mangled, in English way).

I am certain I won't change the name of the GitHub organization, and if the name change is confirmed for `blacksheep`, `rodi`, and `guardpost`, the root namespace will be == to the organization name. The alternative is to keep things like they are, following a scheme that is anyway common in the Python community. Like the [Pallets team](https://github.com/Pallets) (`flask`, `werkzeug` (!!!), `itsdangerous`, `markupsafe`, `jinja`, etc.) or [Encode](https://github.com/Encode) (`uvicorn`, `httpx`, `starlette`, `orm`, `databases`).

If I had enough followers on Twitter, I would have made a poll asking for recommendations, but unfortunately I am nobody on social media.

**Reply by Yourun-proger** [reply]
Understood, good idea, thanks!

**Comment by RobertoPrevato** [comment]
Hi @dolfinus, @Yourun-proger 
I decided to listen to your recommendations, and my brother's recommendation, that matches yours. I will revert the changes to the new packages. Thank you for your feedback!

**Reply by dolfinus** [reply]
That's great, thank you!

**Reply by Yourun-proger** [reply]
Thank you
